### PR TITLE
[Bug] Fixed geo-filter extra layer issue

### LINKED ISSUE #935: [Bug] Geo-filter doesn't work on multiple layers
**Describe the bug**
When adding a new layer to an existing geo-filter, kepler doesn't the map accordingly.

**To Reproduce**
geo-feature-filter-bug

**Expected behavior**
It should update accordingly

### PR DESCRIPTION
**Pull request by macrigiuseppe**
Fix #935 
The geo-filter would not work on layers added after the first one

### Before:

geo-feature-filter-bug

### After:

geo-feature-filter-solution

Signed-off-by: Giuseppe Macri <gmacri@uber.com>

### CODE REVIEW

**Comment by heshan0131** [comment]
you might need to rebase so that my changes aren't included in here

**Comment by macrigiuseppe** [comment]
> you might need to rebase so that my changes aren't included in here

@heshan0131 rebased with master

**Comment by heshan0131** [comment]
> > you might need to rebase so that my changes aren't included in here
> 
> @heshan0131 rebased with master

I still see previous commit, did you `push -f`?

**Comment by macrigiuseppe** [comment]
> > > you might need to rebase so that my changes aren't included in here
> > 
> > 
> > @heshan0131 rebased with master
> 
> I still see previous commit, did you `push -f`?


@heshan0131 yeah, i think i rebased before i merged your changes onto master. 
Rebased again

File: src/reducers/vis-state-updaters.js [hunk 1]
```diff
@@ -585,23 +587,30 @@ export function setFilterUpdater(state, action) {
...
-      // diff filter.dataId with layerDataIds, to determine which dataset to run filter on
-      datasetIds = [
-        ...layerDataIds.filter(lid => !datasetIds.includes(lid)),
-        ...datasetIds.filter(did => !layerDataIds.includes(did))
-      ];
+      // only filter datasetsIds
+      datasetIds = layerDataIds;
+
+      // Update newFilter dataIds
+      const newDataIds = uniq(
```
Comment (heshan0131) [inline src/reducers/vis-state-updaters.js]: simply use `layerDataIds ` seems sufficient, why do you have to recalculate `newDataIds`?

Comment (macrigiuseppe) [reply src/reducers/vis-state-updaters.js]: LayerDataIds is used to determine which datasets to perform the filter against.
The use case is the following:
- add a layer A (dataset_1) to the filter:
  - layerDataIds: [dataset_1]
  - newDataIds: [dataset_1]

- add a layer B (dataset_2) to the filter:
  - layerDataIds: [dataset_2]
  - newDataIds: [dataset_1, dataset_2]

- add a layer C (dataset_1) to the filter:
  - layerDataIds: [dataset_1]
  - newDataIds: [dataset_1, dataset_2]

__layerDataIds__ is necessary to determine which dataset to run the filter against during the current step.
__newDataIds__ is required to update the filter object with. the list of unique dataset ids

**Pull request merged**